contract/contracts/Vesting.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./FanToken.sol";
+
+contract Vesting is Ownable {
+    using SafeMath for uint256;
+
+    mapping(address => bool) public vestingTokens;
+    mapping(address => uint256) public tokensVestingAmount;
+    mapping(address => address) public tokensRecipient;
+    mapping(address => uint256) public tokensVestingStart;
+    mapping(address => uint256) public tokensVestingEnd;
+    mapping(address => uint256) public tokensLastUpdate;
+
+    function addVesting(
```

Comment:
I think it's better to call this function directly from Factory contract. Because the lockup period of the token starts from the point of creation of the token.

Reply:
I intended to use this function like that but I forgot to add calling it on factory contract but only unit tests. I'm gonna add it.a